### PR TITLE
Add docs for library_name

### LINKED ISSUE #264: Add docs about library name
cc @osanseviero @philschmid @julien-c 

There is no `library_name` mentioned in https://huggingface.co/docs/hub/models-cards

The library name is added to the tags of the model on the hub. It's determined as such:

1. The user can force `library_name` in the model card
2. If one of the tags in the model card already is a library name, then do nothing
3. We try to autodetect based on the presence of files for diffusers, keras, sklearn and nemo
4. Lastly, if nothing was detected and there is a valid `config.json` file, the library name is set to `"transformers"`

### PR DESCRIPTION
**Pull request by osanseviero**
Fixes #264

### CODE REVIEW

File: docs/hub/models-cards.md [hunk 1]
```diff
@@ -35,23 +35,41 @@ metrics:
...
+
+2. Having a tag with the name of a library that is supported
 
 ```yaml
 tags:
 - flair
 ```
 
-Find more about our supported libraries [here](./models-libraries), and see the detailed model card specification [here](https://github.com/huggingface/hub-docs/blame/main/modelcard.md).
+If it's not specified, the Hub will try to automatically detect the library type. This approach is discouraged and repo creators should use the explicit license as much as possible.
```
Comment (coyotte508) [inline docs/hub/models-cards.md:64]:
```suggestion
If it's not specified, the Hub will try to automatically detect the library type. This approach is discouraged and repo creators should use the explicit `library_name` as much as possible.
```

**Review by coyotte508 — APPROVED** [review]

File: docs/hub/models-cards.md [hunk 2]
```diff
@@ -35,23 +35,41 @@ metrics:
...
+
+Dataset, metric, and language identifiers are those listed on the [Datasets](https://huggingface.co/datasets), [Metrics](https://huggingface.co/metrics) and [Languages](https://huggingface.co/languages) pages and in the [`datasets`](https://github.com/huggingface/datasets) repository.
+
+See the detailed model card specification [here](https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1).
+
+### Specifying a library
+
+You can also specify the supported libraries in the model card metadata section. Find more about our supported libraries [here](./models-libraries). The library can be specified with the following order of priority
+
+1. Specifying `library_name` in the model card (recommended)
```
Comment (julien-c) [inline docs/hub/models-cards.md:51]:
```suggestion
1. Specifying `library_name` in the model card (recommended if your model is not a `transformers` model)
```

File: docs/hub/models-cards.md [hunk 3]
```diff
@@ -35,23 +35,41 @@ metrics:
...
+
+2. Having a tag with the name of a library that is supported
 
 ```yaml
 tags:
 - flair
 ```
 
-Find more about our supported libraries [here](./models-libraries), and see the detailed model card specification [here](https://github.com/huggingface/hub-docs/blame/main/modelcard.md).
+If it's not specified, the Hub will try to automatically detect the library type. This approach is discouraged and repo creators should use the explicit `library_name` as much as possible.
```
Comment (julien-c) [inline docs/hub/models-cards.md:64]:
```suggestion
If it's not specified, the Hub will try to automatically detect the library type. Unless your model is from `transformers`, this approach is discouraged and repo creators should use the explicit `library_name` as much as possible.
```

**Review by julien-c — COMMENTED** [review]
I don't necessarily agree with the recommendation to always specify `library_name`, notably because 80k out of 120k models are from `transformers` and they won't have a `library_name` in the short/medium term. 

So the current coverage of this metadata item is bound to be low.

As a middleground, I propose the following suggestions. Let me know if this would work for you! 🤟

**Comment by osanseviero** [comment]
Yes, this works for me! Merged your changes

**Review by julien-c — APPROVED** [review]

**Pull request merged**